h2o-genmodel/src/main/java/hex/genmodel/algos/gam/GamMojoModelBase.java

```diff
@@ -132,29 +132,38 @@ int readCatVal(double data, int dataIndex) {
     return eta;
   }
   
+  // check if gamificationis needed.  If all gamified column values are NaN, we need to add gamification.  Otherwise,
+  // gamification is already done.
+  private boolean gamificationNeeded(double[] rawData, int gamColStart) {
+    for (int cind = gamColStart; cind < rawData.length; cind++)
+      if (!Double.isNaN(rawData[cind]))
+        return false;
+    return true;
+  }
+  
   // this method will add to each data row the expanded gam columns
-  public double[] addExpandGamCols(double[] rawData, final RowData rowData) { // add all expanded gam columns here
+  double[] addExpandGamCols(double[] rawData, final RowData rowData) { // add all expanded gam columns here
+    int dataIndEnd = _totFeatureSize - _numExpandedGamCols; // starting index to fill out the rawData
+    if (!gamificationNeeded(rawData, dataIndEnd)) {
+       return rawData;     // already contain gamified columns.  Nothing needs to be done.
```

Comment:
This is definitelly not the fix.
We should throw an exception here because gamified colums should never be provided from the outside, user will never to that. This is just a work-around for testJavaScoring method which based on gam cols being in mojo.names generates extra columns.

Reply:
@honza:
I did not use _names in this fix.  I am just checking if the gamified columns are there.